code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -40,6 +50,7 @@
 	..()
 	if(!targets_from)
 		targets_from = src
+	environment_target_typecache = typecacheof(environment_target_typecache)
```

Comment:
My initial POINT for this change being made was to allow subtypes to choose to ignore sandbags, which before this change, would have been a forced target for all hostile mobs that break surroundings, Hostile mobs are supposed to be customisable, not locked to hardcoded setups.

the fact that typecaches are a thing and speed up that line was just a happy accident to this change.